plugins/catalog-backend/src/service/CatalogBuilder.ts

```diff
@@ -249,6 +251,7 @@ export class CatalogBuilder {
     entitiesCatalog: EntitiesCatalog;
     locationsCatalog: LocationsCatalog;
     higherOrderOperation: HigherOrderOperation;
+    configGenerator: ConfigGenerator;
```

Comment:
I agree that this is turning into a bit of a mess of classes, if it keeps growing for each function that we introduce, but I think it's still the sensible choice at the moment

Reply:
That was my intuition - we will even be able to stub it or block by some feature flags thanks to that separation.